smithy-model/src/main/java/software/amazon/smithy/model/loader/TopologicalShapeSort.java

```diff
@@ -57,7 +65,14 @@ public void enqueue(Shape shape) {
      * @param dependencies Dependencies of the shape.
      */
     public void enqueue(ShapeId shape, Collection<ShapeId> dependencies) {
-        forwardDependencies.put(shape, new LinkedHashSet<>(dependencies));
+        if (dependencies.isEmpty()) {
```

Comment:
I realized we can jump the line and just push straight into satisfiedShapes as an optimization